File: tools/Local Server/controllers/computationTask.js
// controllers/computationTask.js
const logger = require('../utils/logger');

function handleComputationTask(payload, extra) {
    if (!extra.data || !Array.isArray(extra.data)) {
        return JSON.stringify({ success: false, error: 'No data array provided for computation' });
    }

    try {
        let sum = extra.data.reduce((a, b) => a + b, 0);
        return JSON.stringify({ success: true, result: sum });
    } catch (err) {
        logger.error('Error executing computation task: ' + err.message);
        return JSON.stringify({ success: false, error: err.message });
    }
}

module.exports = handleComputationTask;

File: launcher/src/Tasks/ComputationTask.cs
using Newtonsoft.Json.Linq;

namespace CF7Launcher.Tasks
{
    /// <summary>
    /// 数组求和。响应：{success:true, result:<sum>}
    /// </summary>
    public static class ComputationTask
    {
        public static string Handle(JObject message)
        {
            JObject extra = message.Value<JObject>("extra");
            if (extra == null)
                return Error("No extra provided");

            JArray data = extra.Value<JArray>("data");
            if (data == null)
                return Error("No data array provided for computation");

            double sum = 0;
            foreach (JToken item in data)
            {
                sum += item.Value<double>();
            }

            JObject resp = new JObject();
            resp["success"] = true;
            resp["result"] = sum;
            return resp.ToString(Newtonsoft.Json.Formatting.None);
        }

        private static string Error(string msg)
        {
            JObject resp = new JObject();
            resp["success"] = false;
            resp["error"] = msg;
            return resp.ToString(Newtonsoft.Json.Formatting.None);
        }
    }
}
